crates/sui/src/ptb/ptb_parser/argument.rs

```diff
@@ -0,0 +1,145 @@
+                    }
+                }
+                write!(f, "]")
+            }
+            Argument::Option(o) => match o {
+                Some(v) => write!(f, "some({})", v),
+                None => write!(f, "none"),
+            },
+            Argument::ModuleAccess {
+                address,
+                module_name,
+                function_name,
+            } => {
+                write!(f, "{}::{}::{}", address, module_name, function_name)
+            }
+            Argument::TyArgs(ts) => {
+                write!(f, "<")?;
+                for (i, t) in ts.iter().enumerate() {
+                    write!(f, "{}", t.clone().into_type_tag(&resolve_address).unwrap())?;
+                    if i != ts.len() - 1 {
+                        write!(f, ", ")?;
+                    }
+                }
+                write!(f, ">")
+            }
+        }
+    }
+}
+
+impl Argument {
+    pub fn into_move_value_opt(&self) -> Result<MoveValue> {
+        Ok(match self {
+            Argument::Bool(b) => MoveValue::Bool(*b),
+            Argument::U8(u) => MoveValue::U8(*u),
+            Argument::U16(u) => MoveValue::U16(*u),
+            Argument::U32(u) => MoveValue::U32(*u),
+            Argument::U64(u) => MoveValue::U64(*u),
+            Argument::U128(u) => MoveValue::U128(*u),
+            Argument::U256(u) => MoveValue::U256(*u),
+            Argument::Address(a) => MoveValue::Address(a.into_inner()),
+            Argument::Vector(vs) => MoveValue::Vector(
+                vs.iter()
+                    .map(|v| v.into_move_value_opt())
+                    .collect::<Result<Vec<_>>>()
+                    .with_context(|| format!(
+                            "Was unable to parse '{self}'. This is most likely because \
+                            the vector contains non-primitive (i.e., object) values which aren't permitted."
+                    ))?,
+            ),
+            Argument::String(s) => {
+                MoveValue::Vector(s.bytes().into_iter().map(MoveValue::U8).collect::<Vec<_>>())
+            }
+            Argument::Option(o) => {
+                if let Some(v) = o {
+                    let v = v.as_ref().into_move_value_opt().with_context(|| {
+                        format!(
+                            "Was unable to parse '{self}'. This is most likely because \
+                            the option contains a non-primitive (i.e., object) value which isn't permitted."
+                        )
+                    })?;
+                    MoveValue::Vector(vec![v])
+                } else {
+                    MoveValue::Vector(vec![])
+                }
+            }
+            Argument::Identifier(_)
+            | Argument::Array(_)
+            | Argument::ModuleAccess { .. }
+            | Argument::VariableAccess(_, _)
+            | Argument::Gas
+            | Argument::TyArgs(_) => bail!("Was unable to convert '{self}' to primitive value (i.e., non-object value)"),
```

Comment:
How would one fix this? Is this something we expect a user to experience often?

Reply:
This is something that I would hope (expect) to have additional context layered on-top of with `with_context` I think -- but will take a look to make sure we give good messages here -- This shouldn't be too common, but also not "never" either.